airflow/providers/microsoft/azure/hooks/adx.py

```diff
@@ -142,7 +142,9 @@ def warn_if_collison(key, backcompat_key):
                 warnings.warn(
                     f"Conflicting params `{key}` and `{backcompat_key}` found in extras for conn "
                     f"{self.conn_id}. Using value for `{key}`.  Please ensure this is the correct value "
-                    f"and remove the backcompat key `{backcompat_key}`."
+                    f"and remove the backcompat key `{backcompat_key}`.",
+                    UserWarning,
+                    stacklevel=2,
```

Comment:
Just a personal thought that `UserWarning` fit here (and in the same places) better, but maybe I'm wrong

Reply:
I'd agree. Having a concrete warning type is a good thing to have.